src/eval.rs

```diff
@@ -899,6 +961,30 @@ fn apply_continuation<F: LurkField>(
                             store.intern_cont_emit(continuation),
                         );
                     }
+                    Op1::Open => store
+                        .open(*result)
+                        .expect("hidden value could not be opened"),
+                    Op1::Secret => store
+                        .secret(*result)
+                        .expect("secret could not be extracted"),
+                    Op1::Commit => store.hide(F::zero(), *result),
+                    Op1::Num => match result.tag() {
+                        // TODO: There should be a corresponding Op1::Char for creating characters from numbers.
+                        Tag::Num | Tag::Comm | Tag::Char => {
+                            let scalar_ptr =
+                                store.get_expr_hash(result).expect("expr hash missing");
+                            store.intern_num(crate::Num::Scalar::<F>(*scalar_ptr.value()))
+                        }
+                        _ => return Control::Return(*result, *env, store.intern_cont_error()),
```

Comment:
Do we have a unit test for this error? The same question applies to the other errors inserted in this PR.

Reply:
I don't think so. Please go ahead and add circuit/non-circuit tests for such cases when you implement circuit. Much appreciated.